core/org.osate.xtext.aadl2/src/org/osate/xtext/aadl2/validation/Aadl2JavaValidator.java

```diff
@@ -6302,7 +6304,7 @@ else if (destination instanceof Subcomponent && source instanceof Access
 		// Test for L7: connection between subcomponent and access feature of
 		// subcomponent
 		else if (source instanceof Subcomponent && destination instanceof Access
-				&& dstContext instanceof Subcomponent) {
+				&& dstContext instanceof Subcomponent && destinationType != null) {
```

Comment:
There is a better way of doing this instead of a null check. Since `AccessType` is an enumeration, we can use `==` instead of `.equals()` to compare the enum literals. Use `==` for the comparison of `destinationType` to `AccessType.REQUIRES`.